### ISSUE TITLE
Stats saving

### ISSUE THREAD

**Issue by spannerman79** [kangarko/CoreArena#13]
Hi,
Would there be a way to be able to save stats in a database structure?

GodsDead#4614 (discord) originally asked about it in your discord.

Or/and to hook into [LeaderHeads](https://www.spigotmc.org/resources/leaderheads.2079/) support? [API](http://robindebaets.be/leaderheads/api)

**Comment by kangarko** [comment]
Hello!

What kind of statistics do you specifically have in mind?

I have noted the LeaderHeads integration; looks good!

**Comment by spannerman79** [comment]
Not exactly specifics - I'm sure others would like to weigh in here with their suggestions also.

I did a little checking and it seems with LeaderHeads its pretty simple/small the sort of stats that are tracked

They roughly follow;
`nameofplugin`-wins: shows the amount of wins
`nameofplugin`-currency: shows the amount of arena currency (this case nuggets)
`nameofplugin`-games: shows the amount of games
`nameofplugin`-losses: shows the amount of losses

Those are tracked per player, LeaderHeads sorts out the rest (top x, most x, etc)

Basically I looked at https://robindebaets.be/leaderheads/variables `Ctrl-F` "arena" 😛 

GodsDead#4614 (discord) may want more stats to be tracked and even into a separate database (mysql/redis)

----

I don't even mind if this is a premium add-on either.  Take a look at [my Spigot page](https://www.spigotmc.org/members/spanner_man.5170/) and you'll be able to see the premium plugins that I have from you already @kangarko  😄

**Comment by FrozenBeard** [comment]
Alright lets have a think about this!

- Record ALL PVP kills, who killed who, with what, and the name of the item (Data values), and timestamp.
- A way to calculate assists would be very bebeficial, Like in battlefield you get XP for assisting in a kill.
- Record Clan #14 Battles, So Wins/Losses
- Record player Win/Loss stats
- Record how long a battle lasts
- This may be able to be calculated on the fly, but a k/d/a for each player
- Stats should be a central system that can be used between multiple games, so we can create new mini games or events and the data is recorded in that games name, so then we can filter all stats based on the name of that mini-game.
- Record Consumables, so food, potions, end pearls etc, then we can figure out peoples play style and also use this data to see if certain things are too OP for games.

**Comment by spannerman79** [comment]
> It needs to be part of the CoreArena plugin

Not necessarily @FrozenBeard - as no one else has suggested/requested this beforehand it would be safe to assume that the majority don't need/want it.

For those that do - like I stated - even if it was an additional premium add-on I would not mind - it is something that I would like and I am willing to pay/contribute towards it.

I've seen plugins start to have "feature creep" that in turn makes the plugin bloated.  One example is CMI - first it was just to manage player inventory and a couple of other basic functions - now its trying to replace essentials, invsee and about 4 other plugins.


> Record ALL PVP kills, who killed who, with what, and the name of the item (Data values), and timestamp

CoreArena is a battle style arena (PvE) - there are other PvP arena plugins that cover that.

**Comment by FrozenBeard** [comment]
@spannerman79 You must be new to running a community. 
The entire reason players stick around on a server or game is if the have a goal or a way to compete in teams. You are 100% wrong. This is vital core functionality that needs to be added as a core essential to a mini-game. Have you ever seen any giant server with a web UI displaying statistics for games won/lost? Its not a big addition at all, its incrementing values in a database.

> CoreArena is a battle style arena (PvE) where it is the player/s vs Mobs. In fact with Boss it really shines through. The essence of PvE.
> There are other PvP arena plugins that cover that - and it is why i specifically paid for CoreArena - because it is PvE, not PvP.
I completely disagree with you. Its a brand new plugin thats being fleshed out, its called "CoreArena" as in, The Core essentials for Arena battle. I see this as an alternative to BattleArena, which I was very fond of years ago. Its setup process is much better as you can see the outlines of the spawn points and your seleciton phyiscally, no other Arena plugin does this, as you know the plugin litearlly is called Core **ARENA** its not called CorePvE is it. 
You should not try and restrict plugins because you have no use for them, I can see the much bigger picture, I want to create a web UI to display these statistics, I have it planned. Just because you only need something very basic does not mean others do, this is a universal feature that could be adapted in a multitude of ways, even with API connections into the data.

You obviously already have an existing setup that you don't want to move away from.

**Comment by spannerman79** [comment]
> You must be new to running a community.

Incorrect - been doing it since the early 2000's.  Minecraft is just one game.

Don't pretend to walk in my shoes @FrozenBeard

**Comment by Markusi13** [comment]
@FrozenBeard has a great idea in my opinion.
@spannerman79 I wouldn't see why you don't want it to be added, additional features are always great even when you don't have a use for it. Maybe you do later on.

**Comment by LogGits** [comment]
I agree with @FrozenBeard and @Markusi13 I don't see why it shouldn't be added. Just because people don't request something doesn't mean they don't want it. I tried to get LeaderHeads to support CoreArena to which the response was "I'll look into it." @spannerman79 I have plans to use the plugin as a PvP plugin as well as a PvE plugin 😉 

My Post: https://www.spigotmc.org/threads/leaderheads-paid.37008/page-150#post-2699189
Response: https://www.spigotmc.org/threads/leaderheads-paid.37008/page-150#post-2712616

I don't think anyone is complaining about extra features xD It just makes the plugin more appealing to potential buyers and satisfies those who have already purchased the plugin 🙂

**Comment by Owizardo** [comment]
Leaderheads intergration would be awesomesauce!
